debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -586,6 +586,10 @@ public synchronized Connection connection() throws SQLException {
             if (conn == null) throw new SQLException("Unable to obtain a JDBC connection");
             // Always run the initial operations on this new connection
             if (initialOps != null) execute(initialOps);
+            final String statements = config.getString(JdbcConfiguration.ON_CONNECT_STATEMENTS);
+            if (statements != null) {
+                execute(statements.split(";"));
```

Comment:
Should we handle the case where a `;` is given within a string literal?

Reply:
TBH, I tried to postpone the scenario till it happens :-). But we can add a support for escaping - either using backslash or doubling the semicolon. In this case I'd probably prefer doubling the semicolon.